docs/developers/reproducible-contract-builds.md

```diff
@@ -0,0 +1,203 @@
+
+```
+LABEL rust=nightly-2022-03-01
+```
+
+The chosen, _frozen_ image tag should accompany the versioned source code (e.g. via _release notes_), in order to inform others on how to reproduce a specific build (of a specific source code version). In this context, _frozen_ image tag refers to a Docker image tag that will never get any updates after its initial publishing.
+
+:::tip
+If unsure about which one to choose - pick the tag with the **largest index number**. 
+:::
+
+:::warning
+Never pick the tag called `latest` for production-ready builds.
+:::
+
+### Regular builds vs. _deployment_ build
+
+Usually, while developing, testing and integrating a smart contract, one would simply run locally the command `erdpy contract build`, which generates the WASM bytecode. 
+
+However, for production-ready deployments, in order to achieve reproducible builds, make sure to only deploy contracts built as described in the next section, using the chosen Docker image (tag).
+
+## (Re)producing a build
+
+In this section, you'll learn how to run a reproducible build, or, to put it differently, how to reproduce a previous build (made by you or by someone else in the past), on the local machine.
+
+### Fetch the source code
+
+Let's clone the source code locally, and switch to a certain version that we'd like to build:
+
+```
+mkdir -p ~/contracts && cd ~/contracts
+git clone https://github.com/ElrondNetwork/reproducible-contract-build-example.git --branch=v0.1.2 --depth=1
+```
+
+### Preparing environment variables
+
+Let's export the following variables:
+
+```
+export REPOSITORY_ON_HOST=~/contracts/reproducible-contract-build-example
+export REPOSITORY_IN_CONTAINER=/home/elrond/reproducible-contract-build-example
+export IMAGE=elrondnetwork/elrond-sdk-erdpy-rust:frozen-003
+```
+
+The latter export statement explicitly selects the **chosen, _frozen_ Docker image tag** to be used.
+
+:::important
+As of September 2022, you should stick to using the prefix `/home/elrond` for `REPOSITORY_IN_CONTAINER`.
+:::
+
+### Performing the build
+
+Make sure the folder `REPOSITORY_ON_HOST` is accessible from Docker. You can easily do so by granting read & write permissions for all users:
+
+```
+sudo chmod -R a+rw ${REPOSITORY_ON_HOST}
+```
+
+Now let's build the contract by invoking the following `docker run` command:
+
+```
+docker run -it \
+--mount type=bind,source=${REPOSITORY_ON_HOST},destination=${REPOSITORY_IN_CONTAINER} \
+--rm \
+--entrypoint /bin/bash \
+${IMAGE} \
+-c "\
+cd ${REPOSITORY_IN_CONTAINER} && \
+erdpy contract clean --recursive && \
+erdpy contract build --recursive\
+"
+```
+
+After running the command, take ownership over the `output` folders created within `REPOSITORY_ON_HOST`:
+
+```
+sudo chown -R $(id -un):$(id -gn) ${REPOSITORY_ON_HOST}
+```
+
```

Comment:
add a section similar to "you should see the following files in the `output` directory: 
```
├── adder.abi.json
├── adder.imports.json
├── adder.wasm
└── adder.wat
```
"
+ an explanation for each file

Reply:
Added.